### PR TITLE
[@types/stripe] Correct setHttpAgent type

### PR DESCRIPTION
**Pull request by rhodgkins**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/stripe/stripe-node#configuring-a-proxy
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -36,88 +36,95 @@
 
 /// <reference types="node" />
 
+import { Agent } from 'http';
+
 declare class Stripe {
-    DEFAULT_HOST: string;
```
Comment (vaskevich) [inline types/stripe/index.d.ts]: Could you avoid unrelated formatting changes here? This file uses 4 spaces.

Comment (vaskevich) [reply types/stripe/index.d.ts]: On another note, I think this needs an explicit Prettier config.

Comment (rhodgkins) [reply types/stripe/index.d.ts]: I did what the docs said and ran `npm run lint package-name` - this caused the formatting changes?

Comment (vaskevich) [reply types/stripe/index.d.ts]: It hasn't for me in the past - most likely your editor has auto-format on save enabled?

**Comment by rhodgkins** [comment]
@vaskevich I've split up the commits into the change and then the auto-format.

After [committing the Agent change](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38364/commits/ad8fc1ec606540686ff799d19239d18c56ab88b8) the following is outputted from a git commit:

```
git cia "[stripe] Correct setHttpAgent type"
husky > pre-commit (node v8.16.1)
✔  precise-commits: 2 modified file(s) found
✔  [1/2] Processing file: types/stripe/index.d.ts
✔        --> Updated formatting in: types/stripe/index.d.ts
✔  [2/2] Processing file: types/stripe/stripe-tests.ts
ℹ        --> No formatting changes required in: types/stripe/stripe-tests.ts
✔  Formatting complete 🎉
[master ad8fc1ec] [stripe] Correct setHttpAgent type
 2 files changed, 9 insertions(+), 1 deletion(-)
```

You can see it then changes the formatting here, I can get rid of this commit if you prefer but wouldn't it be better to have the formatting correct? Let me know...!
